spartan/devnet-smoke-tests/process_amm_contracts.sh

```diff
@@ -0,0 +1,103 @@
+    token_1_address=$(echo "$contracts_data" | jq -r '.token_1_address')
+    token_liquidity_address=$(echo "$contracts_data" | jq -r '.token_liquidity_address')
+    amm_address=$(echo "$contracts_data" | jq -r '.amm_address')
+
+    private_balance_token_0=$(aztec-wallet simulate balance_of_private -ca "$token_address" --args $current_user_address -f $current_user_address --node-url $NODE_URL | grep "Simulation result:" | awk '{print $3}' | tr -d 'n')
+    private_balance_token_1=$(aztec-wallet simulate balance_of_private -ca "$token_address" --args $current_user_address -f $current_user_address --node-url $NODE_URL | grep "Simulation result:" | awk '{print $3}' | tr -d 'n')
+
+    echo "Account $current_user_address for token 0 address $token_0_address private balance is ${private_balance_token_0}"
+    echo "Account $current_user_address for token 1 address $token_1_address private balance is ${private_balance_token_1}"
+
+    aztec-wallet create-secret -a add-liquidity-nonce
+    aztec-wallet create-authwit transfer_to_public "$amm_address" -ca "$token_0_address" --args "$current_user_address" "$amm_address" "$amount_0_max" secrets:add-liquidity-nonce -f "$current_user_address" -a amm-lp-token-0
+    aztec-wallet create-authwit transfer_to_public "$amm_address" -ca "$token_1_address" --args "$current_user_address" "$amm_address" "$amount_1_max" secrets:add-liquidity-nonce -f "$current_user_address" -a amm-lp-token-1
+
+    amount_0_max=$((private_balance_token_0/4))
+    amount_1_max=$((private_balance_token_1/4))
+    amount_0_min=1
+    amount_1_min=1
+
+    aztec-wallet $([ "$process_proven" = "false" ] && echo "-p native" || echo "") send add_liquidity -ca "$amm_address" --args "$amount_0_max" "$amount_1_max" "$amount_0_min" "$amount_1_min" secrets:add-liquidity-nonce -aw amm-lp-token-0 -aw amm-lp-token-1 -f "$liquidity_provider" $([ "$use_sponsored_fpc" = "true" ] && echo "$SPONSORED_FPC_PAYMENT_METHOD" || echo "")
+
+    ###
+
+    amount_in=$((private_balance_token_0/88))
+
+    aztec-wallet create-secret -a swap-nonce
+    aztec-wallet create-authwit transfer_to_public "$amm_address" -ca "$token_0_address" --args "$current_user_address" "$amm_address" "$amount_in" secrets:swap-nonce -f "$current_user_address" -a amm-swapper-token-0
+
+    amount_out_min=$(aztec-wallet simulate get_amount_out_for_exact_in -ca "$amm_address" --args "$balance_of_public_token_0_amm" "$balance_of_public_token_1_amm" "$amount_in" -f "$swapper")
+    echo "Amount out min for swap: $amount_out_min"
+
+    # TODO: test swap_tokens_for_exact_tokens ?
+    aztec-wallet $([ "$process_proven" = "false" ] && echo "-p native" || echo "") send swap_exact_tokens_for_tokens --ca "$amm_address" --args "$token_0_address" "$token_1_address" "$amount_in" 0 secrets:swap-nonce -aw amm-swapper-token-0 -f "$current_user_address" $([ "$use_sponsored_fpc" = "true" ] && echo "$SPONSORED_FPC_PAYMENT_METHOD" || echo "")
```

Comment:
Note a bunch of things are not being tested at all here, e.g. that we can find the resulting notes. I might be nice to at least check that the token1 balance increased.